mirdata/classes.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+"""
+    TODO all docs
+    """
+
+import numpy as np
+from dataclasses import dataclass
+
+
+# Check consistency
+
+
+def check_types(attributes_names, attributes, expected_type, expected_dtype):
+    for name, attribute, exp_type, int_type in zip(
+        attributes_names, attributes, expected_type, expected_dtype
+    ):
+        if attribute is not None:
+            if not isinstance(attribute, exp_type):
+                raise TypeError(
+                    f"{name} should be a {exp_type}, but is a {type(attribute)} instead"
+                )
+            if not np.asarray(attribute).size == 0:  # check list/ndarray not empty
+                if all(isinstance(type(x), int_type) for x in attribute):
+                    outlier = [
+                        type(x)
+                        for x in attribute
+                        if type(x) is not int_type and type(x) is not None
+                    ]
+                    raise TypeError(
+                        f"{name} should contain only {int_type} elements, but contains an {outlier}"
+                    )
+
+
+def check_lengths(attributes):
+    if len(attributes) > 1:
+        for att1, att2 in zip(attributes[:1], attributes[1:]):
+            if att1 is not None and att2 is not None and not len(att1) == len(att2):
+                raise ValueError("There is a length mismatch between inputs")
+
+
+def check_intervals(intervals):
+    if intervals is not None and not np.shape(intervals)[1] == 2:
+        raise ValueError(
+            f"Intervals should be arrays with two columns, but array has {np.shape(intervals)[1]}"
+        )
+
+
+# Class definitions
+
+
+@dataclass
+class BeatData:
+    """
+    """
+
+    # declare attributes for __repr__
+    times: np.ndarray
+    positions: np.ndarray = None
+
+    # customized __init__ to check data types
+    def __init__(self, times, positions=None):
+        attributes = [times, positions]  # list of attributes
+        attributes_names = ["times", "positions"]
+        expected_type = [np.ndarray, np.ndarray]
+        expected_dtype = [float, int]
+
+        # check mir_eval complience
+        check_types(attributes_names, attributes, expected_type, expected_dtype)
+        check_lengths(attributes)
```

Comment:
Mmmm.. The reason I packed that into functions is because I check that in all classes and it will be a lot of repeated code. Also as we want to add more checks later on I think it would be easy to include functions and call them in the different classes when it makes sense